### ISSUE TITLE
[linux] scaling, SDL? scaling movies etc never happens.

### ISSUE THREAD

**Issue by arrowgent** [alexbatalov/fallout1-ce#15]
```
; Set SCALE_2X=1 to scale the game x2.
; Note: This will increase the minimum resolution to from 640x480 to 1280x960.
SCALE_2X=1

```

this appears to make no change to anything.

```
[MOVIES]

; if MOVIE_SIZE=0 - movies remain their original size.
; if MOVIE_SIZE=1 - movies will stretch to fit the screen while maintaining the aspect ratio of the original.
; if MOVIE_SIZE=2 - movies will stretch to fill the screen.
MOVIE_SIZE=1
```

movies are 640x480 on a 1920x1080 resolution (with black borders)

fo1-ce & fo2-ce

**Comment by alexbatalov** [comment]
These settings are from Sfall which are not yet implemented neither in F2CE, nor in F1CE.

**Comment by sherief** [comment]
I'd love to enable SCALE_2X, if you can provide pointers as to what needs to be done I could give it a shot.

**Comment by a-hurst** [comment]
@arrowgent @sherief This has been fixed/merged by #75 (and also in F2CE). Let me know how it works!

**Comment by a-hurst** [comment]
Okay, just spent some time looking into implementing `MOVIE_SCALE`. My main finding so far is that SDL2 *doesn't support scaling palette-based surfaces*: in order for `SDL_BlitScaled` to work, both the source and destination surface need to be in an RGB or RGBA-type format. With that figured out, I've managed to get a (rough, fairly hacky) version of movie scaling working by defining some new SDL surfaces at the top of the file:

```c
static SDL_Surface* gMovieSdlSurfaceRgb = NULL;
static SDL_Surface* gSdlSurfaceRgb = NULL;
```

and then replacing the last chunk of `movie_MVE_ShowFrame` with this:

```c
    destRect.w = (4 * gSdlSurface->h) / 3;
    destRect.h = gSdlSurface->h;
    destRect.x = (gSdlSurface->w - destRect.w) / 2;
    destRect.y = 0;

    SDL_SetSurfacePalette(surface, gSdlSurface->format->palette);
    gMovieSdlSurfaceRgb = SDL_ConvertSurfaceFormat(surface, SDL_PIXELFORMAT_RGB888, 0);
    gSdlSurfaceRgb = SDL_ConvertSurfaceFormat(gSdlSurface, SDL_PIXELFORMAT_RGB888, 0);
    SDL_BlitScaled(gMovieSdlSurfaceRgb, &srcRect, gSdlSurfaceRgb, &destRect);
    SDL_BlitSurface(gSdlSurfaceRgb, NULL, gSdlTextureSurface, NULL);
    renderPresent();
```

Basically, when drawing every frame, you need to convert the movie frame surface to RGB888 (the pixel format used for F1CE's SDL Renderer), copy it with scaling to another RGB888 surface with the same dimensions asoutput renderer (I just used a copy of `gSdlSurface`), and then copy *that* to the render texture to get drawn to the screen.

I'd be open to making a PR for this, but I'll want to figure out a less messy implementation first. I also don't really understand how the subtitle code plays into the scaling, which will also need to get sorted before a formal PR...
